### PR TITLE
Uf 4536 refactor tags db

### PR DESCRIPTION
**Pull request by josc04**
- Introduced namespace and municipalityId logic for tags
- Added relation between category tags and type tags
- Created repository for status tags
- Created repository for external id type tags
- Created repository for validation settings for tags
- Removed old tags repository

### CODE REVIEW

File: src/main/java/se/sundsvall/supportmanagement/integration/db/model/StatusTagEntity.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		this.modified = modified;
+	}
+
+	public StatusTagEntity withModified(OffsetDateTime modified) {
+		this.modified = modified;
+		return this;
+	}
+
+	@PrePersist
+	void onCreate() {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/integration/db/model/StatusTagEntity.java:135]: Why is the visibility "package private" on this method, but not on onUpdate()?

File: src/main/resources/db/migration/V5_0__change_of_tags_structure.sql [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    created datetime(6),
+    display_name varchar(255),
+    modified datetime(6),
+    municipality_id varchar(255) not null,
+    name varchar(255) not null,
+    namespace varchar(255) not null,
+    primary key (id)
+) engine=InnoDB;
+
+create table external_id_type_tag (
```
Comment (foobar-1337) [inline src/main/resources/db/migration/V5_0__change_of_tags_structure.sql:13]: external_id_type_tag is modeled exactly like status_tag. 
category_tag is almost identical.

Why separate tables?

Comment (josc04) [reply src/main/resources/db/migration/V5_0__change_of_tags_structure.sql:13]: CategoryTag has a relationship to TypeTag while ExternalIdTypeTag and StatusTag does not have any relationships, hence I chose to separate CategoryTag and TypeTag as own entities. I thought about using the same model for ExternalIdTypeTag and StatusTag, but decided to have them as separate tables/repositories. That is of course just my personal taste, I can change the model if desired. Right now the model looks like this (which was verified to be ok with the rest of the team (except you who was on vacation) last week):

image

File: src/main/java/se/sundsvall/supportmanagement/integration/db/model/StatusTagEntity.java (hunk 1 above)
Comment (josc04) [reply src/main/java/se/sundsvall/supportmanagement/integration/db/model/StatusTagEntity.java:135]: Good question, this seem to be implemented on other entities as well. Will change protected to default visibility for all prePresist/preUpdate methods.

File: src/main/resources/db/migration/V5_0__change_of_tags_structure.sql (hunk 2 above)
Comment (foobar-1337) [reply src/main/resources/db/migration/V5_0__change_of_tags_structure.sql:13]: Vacation or not: 
If you see yourself duplicating code, something is wrong with the design.

Comment (josc04) [reply src/main/resources/db/migration/V5_0__change_of_tags_structure.sql:13]: Sure, I'll refactor the model so the StatusTag and the ExternalIdTypeTag shares the same table.

Comment (foobar-1337) [reply src/main/resources/db/migration/V5_0__change_of_tags_structure.sql:13]: Maybe remove suffix?

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**